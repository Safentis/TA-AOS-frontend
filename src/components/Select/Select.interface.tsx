import React from 'react';

export interface Props {
  className?: string;
  options: string[];
  onChange: (event: React.ChangeEvent) => void;
  value: object;
  name: string;
  id: string;
}
