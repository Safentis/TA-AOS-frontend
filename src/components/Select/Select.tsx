import React, { FC } from 'react';

import './Select.css';
import { Props } from './Select.interface';

const Select: FC<Props> = ({
  className = '',
  options,
  onChange,
  value,
  name,
  id,
}): React.ReactElement => {
  return (
    <select
      className={`select ${className}`}
      name={name}
      id={id}
      onChange={onChange}
    >
      {options.map((value: string, index: number) => (
        <option value={value} key={index}>
          {value}
        </option>
      ))}
    </select>
  );
};

export default Select;
