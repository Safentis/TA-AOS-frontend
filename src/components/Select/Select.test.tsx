import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Select from './Select';

import { Props } from './Select.interface';

const setUp = (props: Props) => {
  return shallow(<Select {...props} />);
};

describe('<Select />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {
    className: 'custom-style',
    options: ['one', 'two', 'three'],
    onChange: () => {},
    name: 'select-name',
    value: {},
    id: 'select-id',
  };

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 root select element', () => {
      expect(component.find('select')).toBeTruthy();
    });

    it('has 3 option elements', () => {
      expect(component.find('option').length === 3).toBeTruthy();
    });

    it('has 1 root select element', () => {
      expect(
        component.find('select').hasClass('custom-style'),
      ).toBeTruthy();
    });
  });
});
