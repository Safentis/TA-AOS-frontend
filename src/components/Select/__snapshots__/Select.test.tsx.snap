// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> component does match to the snapshot 1`] = `
<select
  className="select custom-style"
  id="select-id"
  name="select-name"
>
  <option
    key="0"
    value="one"
  >
    one
  </option>
  <option
    key="1"
    value="two"
  >
    two
  </option>
  <option
    key="2"
    value="three"
  >
    three
  </option>
</select>
`;
